packages/chain-adapters/src/utxo/UTXOChainAdapter.ts

```diff
@@ -40,30 +40,35 @@ export interface ChainAdapterArgs {
     ws: bitcoin.ws.Client
   }
   coinName: string
+  coinType: ChainTypes
+  curve?: string
+  bip44Params: BIP44Params
+
 }
 
 export class ChainAdapter implements IChainAdapter<ChainTypes.Bitcoin> {
```

Comment:
This needs to be generic over the chain type actually in use (imagine the type constraint being factored out into some sort of `type UTXOChainTypes = ChainTypes.Bitcoin | ChainTypes.Litecoin | ChainTypes.FoobarCoin` in the future)
```suggestion
export class ChainAdapter<T extends ChainTypes.Bitcoin> implements IChainAdapter<T> {
```